src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSetterTransformer.cs

```diff
@@ -22,21 +19,54 @@ public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode nod
                   && on.Type.GetClrType().FullName == "Avalonia.Styling.Setter"))
                 return node;
 
-            var parent = context.ParentNodes().OfType<XamlAstObjectNode>()
+            IXamlType targetType = null;
+            IXamlLineInfo lineInfo = null;
+
+            var styleParent = context.ParentNodes().OfType<XamlAstObjectNode>()
                 .FirstOrDefault(p => p.Type.GetClrType().FullName == "Avalonia.Styling.Style");
-            
-            if (parent == null)
-                throw new XamlParseException(
-                    "Avalonia.Styling.Setter is only valid inside Avalonia.Styling.Style", node);
-            var selectorProperty = parent.Children.OfType<XamlAstXamlPropertyValueNode>()
-                .FirstOrDefault(p => p.Property.GetClrProperty().Name == "Selector");
-            if (selectorProperty == null)
-                throw new XamlParseException(
-                    "Can not find parent Style Selector", node);
-            var selector = selectorProperty.Values.FirstOrDefault() as XamlIlSelectorNode;
-            if (selector?.TargetType == null)
-                throw new XamlParseException(
-                    "Can not resolve parent Style Selector type", node);
+
+            if (styleParent != null)
```

Comment:
There are conflicts after ControlTheme PR.

Reply:
Conflicts should be fixed now